client/src/components/Attachment/UploadAttachment.js

```diff
@@ -0,0 +1,167 @@
+
+const GQL_CREATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    createAttachment(attachment: $attachment)
+  }
+`
+
+const GQL_UPDATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    updateAttachment(attachment: $attachment)
+  }
+`
+
+const UploadAttachment = ({
+  edit,
+  type,
+  attachmentFunc,
+  setIsAttachment,
+  setLoadingError,
+  relatedObjectUuid
+}) => {
+  const base64Marker = ";base64,"
+  const [error, setError] = useState(null)
+  const [remove, setRemove] = useState(false)
+  const [uploadedList, setUploadedList] = useState([])
+
+  // Trigger attachment submit from report save
+  useEffect(() => {
+    attachmentFunc.current = onSubmit
+  })
+
+  const handleUploadFile = file => {
+    setIsAttachment(true)
+    setUploadedList(current => [...current, file])
+    toast.success("Your document has been uploaded")
+  }
+
+  const handleFileEvent = async e => {
+    const file = e.target.files[0]
+    const selectedAttachment = new Attachment()
+    const base64 = await convertBase64(file)
+
+    // Set initial values for new attachment
+    selectedAttachment.content = base64
+    selectedAttachment.fileName = file.name
+    selectedAttachment.mimeType = base64.split(":").pop().split(";")[0]
+    selectedAttachment.attachmentRelatedObjects[0].relatedObjectType = type
+    selectedAttachment.attachmentRelatedObjects[0].relatedObjectUuid = null
+    selectedAttachment.classification =
+      Settings.fields.attachment.classification.choices.UNDEFINED.value
+
+    save(selectedAttachment).then(response => {
+      selectedAttachment.uuid = response.createAttachment
+      selectedAttachment.contentLength = file.size
+      handleUploadFile(selectedAttachment)
+    })
+  }
+
+  // Convert file to base64 string
+  const convertBase64 = file => {
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader()
+      fileReader.readAsDataURL(file)
+
+      fileReader.onload = () => {
+        resolve(fileReader.result)
+      }
+
+      fileReader.onerror = error => {
+        reject(error)
+      }
+    })
+  }
+
+  return (
+    <div>
+      {error && <Messages error={error} />}
+      {/** **** Select and drop file in here **** **/}
+      <section className="FileUploadContainer">
+        <div>
+          <p className="DragDropText">
+            <Icon className="uploadIcon" size={30} icon={IconNames.EXPORT} />
+            <span style={{ marginTop: "5px" }}>
+              <b>Choose a file</b> or Drag it here
+            </span>
+          </p>
+        </div>
+        <input
+          className="FormField"
+          id="fileUpload"
+          type="file"
+          accept="image/*, .pdf"
+          onChange={handleFileEvent}
+        />
+      </section>
+
+      {/** **** Show uploaded files in here **** **/}
+      <div style={{ display: "flex", flexWrap: "wrap" }}>
+        {uploadedList.map((file, index) => (
```

Comment:
Parameter `file` is better renamed to `attachment`, since it's not a file but an attachment object.